plugins/modules/proxmox_vm_info.py

```diff
@@ -223,10 +225,12 @@ def main():
         module.fail_json(msg="Node %s doesn't exist in PVE cluster" % node)
 
     if not vmid and name:
-        vmid = int(proxmox.get_vmid(name, ignore_missing=False))
+        vmid = proxmox.get_vmid(name, ignore_missing=True)
```

Comment:
Out of curiosity: is it possible that there are multiple such VMs (and thus multiple vmids)?

Just looked at the code of get_vmid: it actually *fails* if there is more than one. How about not using that function, but instead returning the list of all matching VMs?

Reply:
Hmm, you could be right. Proxmox VE allows to create VM with duplicate names. I would rewrite logic to return all VMs with provided name.